integration-cli/docker_api_build_test.go

```diff
@@ -619,6 +619,28 @@ func testBuildWithSession(c *check.C, dir, dockerfile string) (outStr string) {
 	return
 }
 
+func (s *DockerSuite) TestBuildScratchCopy(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+	dockerfile := `FROM scratch
+ADD Dockerfile /
+ENV foo bar`
+	ctx := fakecontext.New(c, "",
+		fakecontext.WithDockerfile(dockerfile),
+	)
+	defer ctx.Close()
+
+	res, body, err := request.Post(
+		"/build",
+		request.RawContent(ctx.AsTarReader(c)),
+		request.ContentType("application/x-tar"))
+	c.Assert(err, checker.IsNil)
+	c.Assert(res.StatusCode, checker.Equals, http.StatusOK)
+
+	out, err := request.ReadBody(body)
+	require.NoError(c, err)
+	assert.Contains(c, string(out), "Successfully built")
```

Comment:
Ideally, we should check here that `OS` was set. Do you know why there is an override in inspect that currently prevents it https://github.com/moby/moby/blob/b00b1b1c40750dd0239fb30e421686b63b4183bd/daemon/image_inspect.go#L22-L25. That doesn't look correct. (cc @jhowardmsft )

Reply:
@tonistiigi It's removed in the PR to re-coalesce the stores. https://github.com/moby/moby/pull/34859/files#diff-1ed32f2ce4c90e536bbbb9ede729120cL21